pkg/dockerregistry/server/blobdescriptorservice.go

```diff
@@ -204,7 +211,8 @@ func imageHasBlob(
 	}
 
 	// only manifest V2 schema2 has docker image config filled where dockerImage.Metadata.id is its digest
-	if len(image.DockerImageConfig) > 0 && image.DockerImageMetadata.ID == blobDigest {
+	if image.DockerImageManifestMediaType == schema2.MediaTypeManifest &&
```

Comment:
That's right. `image.DockerImageManifestMediaType == schema2.MediaTypeManifest` is equivalent to `len(image.DockerImageConfig) > 0`.